### PR TITLE
Add unit test in Github action CI.

### LINKED ISSUE #78: Fix CI
* Integrate running unit tests on PR/merge

### PR DESCRIPTION
**Pull request by arijitAD**

### CODE REVIEW

File: .github/workflows/on-pr.yml [hunk 1]
```diff
@@ -10,3 +10,57 @@ jobs:
...
+
+      - name: Checkout code
+        uses: actions/checkout@v2
+
+      - name: Start database
+        run:  docker-compose -f docker-compose.yml up -d db
+
+      - name: Run unit tests
+        run:
+          PGPASSWORD=password make statedifftest
```
Comment (ashwinphatak) [inline .github/workflows/on-pr.yml]: The password for test db is also in the Makefile. Why do we need it here?

File: .github/workflows/on-pr.yml [hunk 2]
```diff
@@ -10,3 +10,57 @@ jobs:
...
+    name: Run state diff unit test
+    env:
+      GOPATH: /tmp/go
+    strategy:
+      matrix:
+        go-version: [ 1.15.x]
+        platform: [ubuntu-latest]
+    runs-on: ${{ matrix.platform }}
+    steps:
+      - name: Create GOPATH
```
Comment (ashwinphatak) [inline .github/workflows/on-pr.yml:50]: The first three steps here are the same as the earlier job. Can't we factor this out and run the tests in the same folder?

File: .github/workflows/on-pr.yml (hunk 1 above)
Comment (arijitAD) [reply .github/workflows/on-pr.yml]: Added this into Makefile as env.

File: .github/workflows/on-pr.yml (hunk 2 above)
Comment (arijitAD) [reply .github/workflows/on-pr.yml:50]: Yes, We can but then the tests will run sequentially in one job.
Currently, we are running all the tests in parallel hence the setup steps need to be done for each individual test.

Comment (arijitAD) [reply .github/workflows/on-pr.yml:50]: Let me explore this a bit. I think we can parallelize the job after the initial setup.

**Review by ashwinphatak — APPROVED** [review]

**Review by i-norden — APPROVED** [review]
LGTM!

**Pull request merged**